server/org_config.go

```diff
@@ -0,0 +1,187 @@
+	orgID, ok := hasOrganizationContext(ctx)
+	if !ok {
+		Error(w, http.StatusBadRequest, "Organization not found on context", s.Logger)
+		return
+	}
+
+	config, err := s.Store.OrganizationConfig(ctx).FindOrCreate(ctx, orgID)
+	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+
+	if config == nil {
+		Error(w, http.StatusBadRequest, "Organization configuration object was nil", s.Logger)
+		return
+	}
+	res := newOrganizationConfigResponse(*config)
+
+	encodeJSON(w, http.StatusOK, res, s.Logger)
+}
+
+// LogViewerOrganizationConfig retrieves the log viewer UI section of the organization config
+func (s *Service) LogViewerOrganizationConfig(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	orgID, ok := hasOrganizationContext(ctx)
+	if !ok {
+		Error(w, http.StatusBadRequest, "Organization not found on context", s.Logger)
+		return
+	}
+
+	config, err := s.Store.OrganizationConfig(ctx).FindOrCreate(ctx, orgID)
+	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+
+	if config == nil {
+		Error(w, http.StatusBadRequest, "Organization configuration object was nil", s.Logger)
+		return
+	}
+
+	res := newLogViewerConfigResponse(*config)
+
+	encodeJSON(w, http.StatusOK, res, s.Logger)
+}
+
+// ReplaceLogViewerOrganizationConfig replaces the log viewer UI section of the organization config
+func (s *Service) ReplaceLogViewerOrganizationConfig(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	orgID, ok := hasOrganizationContext(ctx)
+	if !ok {
+		Error(w, http.StatusBadRequest, "Organization not found on context", s.Logger)
+		return
+	}
+
+	var logViewerConfig chronograf.LogViewerConfig
+	if err := json.NewDecoder(r.Body).Decode(&logViewerConfig); err != nil {
+		invalidJSON(w, s.Logger)
+		return
+	}
+	if err := validLogViewerConfig(logViewerConfig); err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+
+	config, err := s.Store.OrganizationConfig(ctx).FindOrCreate(ctx, orgID)
+	if err != nil {
+		Error(w, http.StatusBadRequest, err.Error(), s.Logger)
+		return
+	}
+	if config == nil {
+		Error(w, http.StatusBadRequest, "Organization configuration object was nil", s.Logger)
+		return
+	}
+	config.LogViewer = logViewerConfig
+
+	res := newLogViewerConfigResponse(*config)
+	if err := s.Store.OrganizationConfig(ctx).Update(ctx, config); err != nil {
+		unknownErrorWithMessage(w, err, s.Logger)
+		return
+	}
+
+	encodeJSON(w, http.StatusOK, res, s.Logger)
+}
+
+// validLogViewerConfig ensures that the request body log viewer UI config is valid
+// to be valid, it must: not be empty, have at least one column, not have multiple
+// columns with the same name or position value, each column must have a visbility
+// of either "visible" or "hidden" and if a column is of type severity, it must have
+// at least one severity format of type icon, text, or both
+func validLogViewerConfig(cfg chronograf.LogViewerConfig) error {
```

Comment:
Are there any tests for this function?

Reply:
There are no explicit tests for this method, we could add them

Reply:
added.